src/search.py

```diff
@@ -0,0 +1,148 @@
+from typing import Dict
```

Comment:
You could add a docstring for the module with an "example usage" section that demonstrates what the config should contain?